src/js/controller/controller.js

```diff
@@ -311,19 +314,24 @@ define([
                 }
 
                 _actionOnAttach = _completeHandler;
-                if (_model.get('repeat')) {
-                    _next();
-                } else {
-                    if (_model.get('item') === _model.get('playlist').length - 1) {
+
+                var idx = _model.get('item');
+                if (idx === _model.get('playlist').length - 1) {
+                    // If it's the last item in the playlist
+                    if (_model.get('repeat')) {
+                        _next();
+                    } else {
                         _loadOnPlay = 0;
-                        _stop(true);
                         setTimeout(function() {
```

Comment:
does this still need to be deferred? We should try to get rid of as many of these as possible.